landoapi/transplants.py

```diff
@@ -301,6 +303,47 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
+    if short_name == "mozilla-central":
+        message = ""
+        try:
+            product_details = requests.get(
+                "https://product-details.mozilla.org/1.0/firefox_versions.json"
+            ).json()
+
+            if "NEXT_SOFTFREEZE_DATE" not in product_details:
+                raise KeyError
```

Comment:
This is redundant, as `product_details["NEXT_SOFTFREEZE_DATE"]` will raise `KeyError` if the key does not exist. Either return a different value or handle this differently.